cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -322,16 +323,22 @@ def ensure_routes(vpc_conn, route_table, route_specs, propagating_vgw_ids,
 
     changed = routes_to_delete or route_specs_to_create
     if changed:
-        for route_spec in route_specs_to_create:
-            vpc_conn.create_route(route_table.id,
-                                  dry_run=check_mode,
-                                  **route_spec)
-
-        for route in routes_to_delete:
-            vpc_conn.delete_route(route_table.id,
-                                  route.destination_cidr_block,
-                                  dry_run=check_mode)
-    return {'changed': changed}
+        try:
+            for route_spec in route_specs_to_create:
+                    vpc_conn.create_route(route_table.id,
+                                          dry_run=check_mode,
+                                          **route_spec)
+
+            for route in routes_to_delete:
+                vpc_conn.delete_route(route_table.id,
+                                      route.destination_cidr_block,
+                                      dry_run=check_mode)
+
+        except EC2ResponseError as e:
+            if e.error_code == 'DryRunOperation':
+                pass
+
+    return {'changed': len(changed) > 0}
 
```

Comment:
I realize now you meant you didn't see a reason for the rest of the changes.  I'll go through and add comments for those.